### PR TITLE
Update deploy things for py312 + OpenID Connect publishing

### PR DESCRIPTION
**Pull request by IAlibay**
Towards #4326 

Title says it all

Changes made in this Pull Request:
 - cibuildwheel runners are all updated
 - updates various other actions being called

## Developers certificate of origin
- [x] I certify that this contribution is covered by the LGPLv2.1+ license as defined in our [LICENSE](https://github.com/MDAnalysis/mdanalysis/blob/develop/LICENSE) and adheres to the [**Developer Certificate of Origin**](https://developercertificate.org/).

----
:books: Documentation preview :books:: https://mdanalysis--4325.org.readthedocs.build/en/4325/

### CODE REVIEW

**Comment by IAlibay** [comment]
OpenID Connect trusted publisher for test and normal PyPi have now been set for MDAnalysis and MDAnalysisTests

**Comment by IAlibay** [comment]
For the sake of security, all old API tokens have also been deleted.

File: .github/workflows/deploy.yaml
Comment (orbeckst) [inline .github/workflows/deploy.yaml]: What is now happening (compared to previously) when we get a pull request against develop?

File: .github/workflows/deploy.yaml
Comment (orbeckst) [inline .github/workflows/deploy.yaml:172]: Is the MDA deployment so complicated that we can't use our action https://github.com/MDAnalysis/pypi-deployment ?

**Review by orbeckst — APPROVED** [review]
I hope it works! Thanks for the update.

(Small question about including "develop" but I assume that this fine.)

File: .github/workflows/deploy.yaml
Comment (IAlibay) [reply .github/workflows/deploy.yaml:172]: cibuildwheels deployments don't work via pypi-deployment right now - that's a bit more complicated so I've not added that branch yet (it's a bit more messy because you usually want to parallelise at the level of runners for each flavour you build wheels for - you don't fully have to though, you can just trust the docker container to get it right).

One day I'll get that done, it's just not been a burning priority (although the amount of cython packages we're maintinaing is starting to grow).

What I can't stuff in a reusable action though is the arm builds (using Cirrus), those will need to exist everywhere we need those builds (currently just here).

File: .github/workflows/deploy.yaml
Comment (IAlibay) [reply .github/workflows/deploy.yaml]: This is just a test thing for this PR, I'm removing it before merge.

File: .github/workflows/deploy.yaml
Comment (orbeckst) [reply .github/workflows/deploy.yaml:172]: Thanks for the explanations!

Comment (IAlibay) [reply .github/workflows/deploy.yaml:172]: Extra info for anyone reading this and going "but why cibuildwheels here and not everywhere else" (it's a recurring question we should document) - we use cibuildwheels to build anything that can't do a pure Python wheel (i.e. uses Cython).

Comment (orbeckst) [reply .github/workflows/deploy.yaml:172]: Do we have a "maintainer FAQ" somewhere — wiki or NF drive?

Comment (IAlibay) [reply .github/workflows/deploy.yaml:172]: Unfortunately not, we have barbones deployment docs on the userguide but yeah there's just not been enough of my time to add docs for everything - that's what we applied for that SDG for though so fingers crossed!

**Pull request merged**
